### ISSUE TITLE
Program crash when switching between songbooks.

### ISSUE THREAD

**Issue by vladozar** [SoftProjector/softprojector#45]
switch between songbooks the program crashes completely and end up having to open it again

**Comment by vladozar** [comment]
I have spent some time debugging this issue, I cannot replicate it. If anyone will come across this again, Please provide detailed steps how to reproduce it.

**Comment by Dmitrix83** [comment]
Which version are you using? we are using the master (2.2) version on
github compile with QT.
How many databases of song book you have imported? We have about 10

We had it crash few times where we have selected one song book did some
search.. and then select all song books and try to search immediately it
would lock up..

This maybe not the right place.. but when you add video media to the
schedule and then view few songs and bible text. then double click on media
under schedule windows software also locks up..

**Comment by vladozar** [comment]
I'm running the latest source on the master branch.  Qt 5.12.6, Windows 10.
Have tried both 32 and 64 bit versions. Added more than 10 songbooks. Still not able to reproduce.

**Comment by Dmitrix83** [comment]
For program crash when searching songbooks..
Try this..
Select a song book from the list.. start search "слушайте" for example  and
then go to selection and select "All songbooks"  and this is when
application will crash for me..

and from Qt application output
QFont::fromString: Invalid description '(empty)'
..
..
QFont::fromString: Invalid description ',28,-1,5,50,0,0,0,0,0'
QFont::fromString: Invalid description '(empty)'
..
..
QSortFilterProxyModel: index from wrong model passed to mapFromSource
...
...
08:29:18: The program has unexpectedly finished.
08:29:18: The process was ended forcefully.

**Comment by vladozar** [comment]
Yeah, I was able to reproduce the issue a while ago now, but just got back to trying to figure out what is going on. The issue is with `QSortFilterProxyModel: index from wrong model passed to mapFromSource` I'm not sure yet what the exact issue it, but working on it now.

**Comment by vladozar** [comment]
Ok, after hours of debugging and looking at the wrong place, I finally was able to fix the issue. A PR was created, will be glad if anyone reviews it. 
https://github.com/SoftProjector/softprojector/pull/50
